test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -454,17 +458,73 @@ public void GetUriByAddress_WithHttpContext_IncludesAmbientValues()
 
             var linkGenerator = CreateLinkGenerator(endpoint1, endpoint2);
 
-            var httpContext = CreateHttpContext(new { controller = "Home", });
+            var httpContext = CreateHttpContext();
             httpContext.Request.Scheme = "http";
             httpContext.Request.Host = new HostString("example.com");
 
             // Act
-            var uri = linkGenerator.GetUriByAddress(httpContext, 1, values: new RouteValueDictionary(new { action = "Index", }));
+            var uri = linkGenerator.GetUriByAddress(
+                httpContext, 
+                1, 
+                values: new RouteValueDictionary(new { action = "Index", }),
+                ambientValues: new RouteValueDictionary(new { controller = "Home", }));
 
             // Assert
             Assert.Equal("http://example.com/Home/Index", uri);
         }
 
+        [Fact]
+        public void GetPathByAddress_WithHttpContext_CanOverrideUriParts()
+        {
+            // Arrange
+            var endpoint1 = EndpointFactory.CreateRouteEndpoint("{controller}/{action}/{id}", metadata: new object[] { new IntMetadata(1), });
+            var endpoint2 = EndpointFactory.CreateRouteEndpoint("{controller}/{action}/{id?}", metadata: new object[] { new IntMetadata(1), });
+
+            var linkGenerator = CreateLinkGenerator(endpoint1, endpoint2);
+
+            var httpContext = CreateHttpContext();
+            httpContext.Request.Scheme = "http";
+            httpContext.Request.Host = new HostString("example.com");
+            httpContext.Request.PathBase = "/Foo";
```

Comment:
Do schema and host have any effect or are they just required for the test to pass?

Reply:
Yeah, bad copy paste they don't do anything here. Removed.